src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs

```diff
@@ -499,6 +495,56 @@ private KeyedByTypeCollection<IOperationBehavior> GetIOperationBehaviorAttribute
             }
         }
 
+        // Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+        // that matches in name and parameter types.  Null is returned if there is no match.
+        private static MethodInfo GetCorrespondingMethodFromType(Type type, MethodInfo methodInfo)
+        {
+            Contract.Assert(type != null);
+            Contract.Assert(methodInfo != null);
+
+            if (methodInfo.DeclaringType == type)
+            {
+                return methodInfo;
+            }
+
+            MethodInfo matchingMethod = type.GetTypeInfo().DeclaredMethods.SingleOrDefault(m => MethodsMatch(m, methodInfo));
```

Comment:
You want to enumerate all methods. If there's some ambiguity, we want to catch that bug.